### PR TITLE
Rename .titan extension and directory titan-compiler

### PR DESCRIPTION
**Pull request by pjpollina**
As per the 3rd and 4th objective of Issue #22

### CODE REVIEW

**Comment by hugomg** [comment]
Thanks for showing interest in helping us with this issue.

Please let us know if you need assistance in getting the test suite to run on your machine. It has a shorter feedback loop than Travis :)

**Comment by pjpollina** [comment]
That'd be great, yeah. I really wanna know why this keeps failing.

**Comment by hugomg** [comment]
1. First of all, you need to make sure that you have the [Luarocks](https://luarocks.org/) package manager installed on your machine.

2. To be able to run the pallene compiler you need to install its dependencies:

```sh
luarocks install --local --only-deps pallene-dev-1.rockspec
```

3. You also need to be able to build the runtime library and our modified version of Lua 5.4. Currently our makefile only works on Linux and Mac. If on Linux, make sure that you have the dev headers for the readline library installed (sudo apt install readline-dev, or equivalent for your distro)

```sh
make linux-readline  # for linux
make macosx          # for macos
```

4. Assuming I wrote everything correctly here, you should now be able to run Pallene. (Do all of these from the root project directory, as per bug #16).

```sh
# running the compiler
./pallenec foo.pallene
./lua/src/lua -l foo
```

5. To run the test suite you need to install [Busted](http://olivinelabs.com/busted/). You should be able to do so with Luarocks:

```sh
luarocks install --local busted
```

6. Run busted to run the full test suite (or just one of the tests)

```sh
busted                       # run all tests
busted spec/parser_spec.lua  # run just one of the test files
```

-----

Please let me know if this works for you or not. If it does, I'll fix our README file to use these instructions.

BTW, you don't need to close and reopen this PR all the time. Travis already informs us when the PR is broken and needs to be fixed so you don't need to do that yourself.

**Comment by pjpollina** [comment]
Thanks! I got this pull working but I'll definitely use that for my future contributions :D

**Comment by pjpollina** [comment]
Sure thing!

File: pallene/driver.lua [hunk 1]
```diff
@@ -103,8 +103,8 @@ end
 -- [output_ext]. Erases any intermediate files that are produced along the way.
 --
 -- Example:
---    compile("titan", "so", "foo.titan") --> outputs "foo.so"
---    compile("titan", "c", "foo.titan")  --> outputs "foo.c"
+--    compile("titan", "so", "foo.pallene") --> outputs "foo.so"
+--    compile("titan", "c", "foo.pallene")  --> outputs "foo.c"
```
Comment (hugomg) [inline pallene/driver.lua]: These need to become `compile("pallene", ...` as well

Comment (pjpollina) [reply pallene/driver.lua]: Got it, will do.

**Comment by pjpollina** [comment]
Think I got them all.

**Comment by hugomg** [comment]
Thanks!

By the way, have you tried running the test suite on your side? I'm rewriting the README right now so I'd love if you could confirm if the instructions worked.

**Pull request merged**